### DISCUSSION TITLE
[Done] Range adjustement by band

### DISCUSSION THREAD

**Discussion by emmanuelmathot** [developmentseed/titiler#304] [Feature Request]
Today, titiler API allows to rescale values from a min and max for all the bands values. Would it be possible to submit an array of ranges corresponding to each bands selected to make the RGB?
something like
```
assets=swir22,nir,red&rescale=[[0,5000],[0.7000],[0,10000]]
```

**Comment by vincentsarago** [comment]
It's not document this is already possible thanks to rio/tiler 
If you pass something like 
`rescale=1,100,1,1000,1,2555` rio-tiler will split the array for each band (I think) 

This is not a great way to define it 
I would prefer something like 
`rescale="1,1000"&rescale="1,100"&rescale="1,255"` which seems a better way to pass multiple parameters 🤷‍♂️

**Reply by emmanuelmathot** [reply]
Great! Thank you for the answer. So I understand we should test with `rescale=1,100,1,1000,1,2555` in the current implementation and it should work.

**Reply by emmanuelmathot** [reply]
I confirm it works! Thx @vincentsarago

**Reply by vincentsarago** [reply]
FYI https://github.com/cogeotiff/rio-tiler/blob/6c8da5e88ca562205998fae73aa42a0e9a0014b5/rio_tiler/models.py#L218-L228 is where the `magic` happens

**Comment by vincentsarago** [comment]
@emmanuelmathot FYI this is going to change in next titiler version per https://github.com/developmentseed/titiler/pull/319

As I mentioned in https://github.com/developmentseed/titiler/discussions/304#discussioncomment-659609 we are going to move to `rescale=1,1000&rescale=1,100&rescale=1,255` which seems to be a better way to define multiple values.

**Reply by emmanuelmathot** [reply]
Thank you for the heads up. Is the same logic applicable for `nodata` param?

**Reply by vincentsarago** [reply]
nodata is a unique value! 

The other multiple parameter is `bidx` which is provided via `coma (,) delimited sting` (e.g: "1,2,3"). We could switch to `bidx=1&bidx=2&bidx=3` but this might impact much more people so I'll need to first start the conversation before switching this
